### PR TITLE
Refactor toDataFrame implementation in compiler plugin

### PR DESCRIPTION
**Pull request by koperagen**

### CODE REVIEW

File: plugins/kotlin-dataframe/src/org/jetbrains/kotlinx/dataframe/plugin/loadInterpreter.kt [hunk 1]
```diff
@@ -161,6 +164,9 @@ internal inline fun <reified T> String.load(): T {
         "ReadDelimStr" -> ReadDelimStr()
         "GroupByToDataFrame" -> GroupByToDataFrame()
         "ToDataFrameFrom0" -> ToDataFrameFrom()
```
Comment (zaleslaw) [inline plugins/kotlin-dataframe/src/org/jetbrains/kotlinx/dataframe/plugin/loadInterpreter.kt:166]: To and to?, again, could they be a CONSTANTS?

File: plugins/kotlin-dataframe/src/org/jetbrains/kotlinx/dataframe/plugin/interpret.kt [hunk 2]
```diff
@@ -269,6 +270,10 @@ fun <T> KotlinTypeFacade.interpret(
     }
 }
 
+fun interpretationFrameworkError(message: String): Nothing = throw InterpretationFrameworkError(message)
```
Comment (zaleslaw) [inline plugins/kotlin-dataframe/src/org/jetbrains/kotlinx/dataframe/plugin/interpret.kt:273]: the name looks like object name, not a function

File: plugins/kotlin-dataframe/src/org/jetbrains/kotlinx/dataframe/plugin/interpret.kt [hunk 3]
```diff
@@ -89,13 +89,14 @@ fun <T> KotlinTypeFacade.interpret(
...
     if (conflictingKeys.isNotEmpty()) {
-        error("Conflicting keys: $conflictingKeys")
+        interpretationFrameworkError("Conflicting keys: $conflictingKeys")
     }
     val expectedArgsMap = processor.expectedArguments
         .filterNot { it.name.startsWith("typeArg") }
         .associateBy { it.name }.toSortedMap().minus(additionalArguments.keys)
 
-    if (expectedArgsMap.keys - defaultArguments != actualArgsMap.keys - defaultArguments) {
+    val unexpectedArguments = expectedArgsMap.keys - defaultArguments != actualArgsMap.keys - defaultArguments
```
Comment (zaleslaw) [inline plugins/kotlin-dataframe/src/org/jetbrains/kotlinx/dataframe/plugin/interpret.kt:98]: very interesting algebra of arguments (great to explain somehow this kind of logic)

File: plugins/kotlin-dataframe/src/org/jetbrains/kotlinx/dataframe/plugin/analyzeRefinedCallShape.kt [hunk 4]
```diff
@@ -38,44 +32,6 @@ fun KotlinTypeFacade.analyzeRefinedCallShape(call: FirFunctionCall, reporter: In
...
-                    toDataFrame(maxDepth, call, TraverseConfiguration())
-                } else {
-                    PluginDataFrameSchema(emptyList())
-                }
-            }
-            "toDataFrameDefault" -> {
-                val maxDepth = 0
-                toDataFrame(maxDepth, call, TraverseConfiguration())
-            }
             "Aggregate" -> {
```
Comment (zaleslaw) [inline plugins/kotlin-dataframe/src/org/jetbrains/kotlinx/dataframe/plugin/analyzeRefinedCallShape.kt:35]: sorry, again better to have CONSTANTS, believe that these names are repeated in codebase

**Review by zaleslaw — APPROVED** [review]

File: plugins/kotlin-dataframe/src/org/jetbrains/kotlinx/dataframe/plugin/interpret.kt (hunk 2 above)
Comment (koperagen) [reply plugins/kotlin-dataframe/src/org/jetbrains/kotlinx/dataframe/plugin/interpret.kt:273]: Can't argue, but i was inspired by "error" function from standard library

File: plugins/kotlin-dataframe/src/org/jetbrains/kotlinx/dataframe/plugin/interpret.kt (hunk 3 above)
Comment (koperagen) [reply plugins/kotlin-dataframe/src/org/jetbrains/kotlinx/dataframe/plugin/interpret.kt:98]: This check is needed to simplify debug if someone writes an interpreter for `fun a(myArgument: String)` but misspells myArgument as myArg123123ument. Expected argument (by interpreter) is myArg123123ument, actual myArgument.
Easy to check.
But with default value:
fun a(myArgument: String, myArgument1: String = "defaultValue")
a("abc")
actualArg = myArgument, expectedArg = myArg123123ument, myArgument1.
We know that for myArgument1 default value is provided, so we exclude such arguments from comparison: it's not a mistake.

File: plugins/kotlin-dataframe/src/org/jetbrains/kotlinx/dataframe/plugin/analyzeRefinedCallShape.kt (hunk 4 above)
Comment (koperagen) [reply plugins/kotlin-dataframe/src/org/jetbrains/kotlinx/dataframe/plugin/analyzeRefinedCallShape.kt:35]: Answered in other PR :)

**Pull request merged**